cmd/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/SPSCommerce/kube-hpa-scale-to-zero/internal"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.uber.org/zap/zapcore"
+	"golang.org/x/net/context"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/metrics/pkg/client/custom_metrics"
+	"k8s.io/metrics/pkg/client/external_metrics"
+	"net/http"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"time"
+)
+
+type RunConfiguration struct {
+	KubeConfigPath    string
+	HpaSelector       string
+	Port              int
+	IsDevelopmentMode bool
+}
+
+func loadConfiguration() *RunConfiguration {
+	config := &RunConfiguration{
+		KubeConfigPath:    "",
+		HpaSelector:       "scaletozero.spscommerce.com/watch",
+		Port:              443,
+		IsDevelopmentMode: false,
+	}
+
+	// how to run service itself
+	flag.StringVar(&config.KubeConfigPath, "kube-config", config.KubeConfigPath, "(optional) kube config to use when ")
+	flag.BoolVar(&config.IsDevelopmentMode, "development-mode", config.IsDevelopmentMode, fmt.Sprintf("(optional) turn on development mode for logs"))
+	flag.IntVar(&config.Port, "port", config.Port, "(optional) app port")
+	flag.StringVar(&config.HpaSelector, "hpa-selector", config.HpaSelector, "(optional) hpa label to watch for")
+
+	flag.Parse()
+
+	return config
+}
+
+func UpEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	_, err := w.Write([]byte("happy"))
+	if err != nil {
+		ctrl.Log.Error(err, "unable to handle request to up endpoint")
+	}
+}
+
+func main() {
+
+	runtimeConfig := loadConfiguration()
+
+	opts := zap.Options{
+		Development: runtimeConfig.IsDevelopmentMode,
```

Comment:
If you are substituting zap with go-logr in most places why not do that in here as well?

Reply:
we go-logr is an interface to declare the contract, but we still use zap as implementation.